nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -133,69 +114,30 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     boolean newTable = base == null;
     String newMetadataLocation = writeNewMetadataIfRequired(newTable, metadata);
 
-    String refName = client.refName();
     boolean failure = false;
     try {
       String contentId = table == null ? null : table.getId();
       client.commitTable(base, metadata, newMetadataLocation, contentId, key);
-    } catch (NessieConflictException ex) {
-      failure = true;
-      if (ex instanceof NessieReferenceConflictException) {
-        // Throws a specialized exception, if possible
-        maybeThrowSpecializedException((NessieReferenceConflictException) ex);
+    } catch (NessieConflictException | NessieNotFoundException | HttpClientException ex) {
+      if (ex instanceof NessieConflictException || ex instanceof NessieNotFoundException) {
+        failure = true;
```

Comment:
I think it might be preferable to delegate determining the `failure` flag to `NessieUtil.handleExceptionsForCommits`. This logic should be the same between tables and views.

Perhaps `handleExceptionsForCommits` could return an object with both the flag and the exception to be re-thrown. WDYT?

Reply:
That will be more complicated I guess. 

I have simplified a bit now. Please take a look